src/v/pandaproxy/schema_registry/sharded_store.cc

```diff
@@ -0,0 +1,227 @@
+                   sub,
+                   version,
+                   inc_del))
+                  .value();
+    auto s = co_await get_schema(v_id.id);
+
+    co_return subject_schema{
+      .sub = sub,
+      .version = v_id.version,
+      .id = v_id.id,
+      .type = s.type,
+      .definition = std::move(s.definition),
+      .deleted = v_id.deleted};
+}
+
+ss::future<std::vector<subject>>
+sharded_store::get_subjects(include_deleted inc_del) {
+    auto map = [inc_del](store& s) { return s.get_subjects(inc_del); };
+    auto reduce = [](std::vector<subject> acc, std::vector<subject> subs) {
+        acc.insert(
+          acc.end(),
+          std::make_move_iterator(subs.begin()),
+          std::make_move_iterator(subs.end()));
+        return acc;
+    };
+    co_return co_await _store.map_reduce0(map, std::vector<subject>{}, reduce);
+}
+
+ss::future<std::vector<schema_version>>
+sharded_store::get_versions(const subject& sub, include_deleted inc_del) {
+    auto versions = co_await _store.invoke_on(
+      shard_for(sub), &store::get_versions, sub, inc_del);
+    co_return std::move(versions).value();
+}
+
+ss::future<std::vector<schema_version>>
+sharded_store::delete_subject(const subject& sub, permanent_delete permanent) {
+    auto versions = co_await _store.invoke_on(
+      shard_for(sub), &store::delete_subject, sub, permanent);
+    co_return std::move(versions).value();
+}
+
+ss::future<bool> sharded_store::delete_subject_version(
+  const subject& sub,
+  schema_version version,
+  permanent_delete permanent,
+  include_deleted inc_del) {
+    auto deleted = co_await _store.invoke_on(
+      shard_for(sub),
+      &store::delete_subject_version,
+      sub,
+      version,
+      permanent,
+      inc_del);
+    co_return deleted.value();
+}
+
+ss::future<compatibility_level> sharded_store::get_compatibility() {
+    co_return _store.local().get_compatibility().value();
+}
+
+ss::future<compatibility_level>
+sharded_store::get_compatibility(const subject& sub) {
+    using overload_t = result<compatibility_level> (store::*)(const subject&)
+      const;
+    auto level = co_await _store.invoke_on(
+      shard_for(sub), static_cast<overload_t>(&store::get_compatibility), sub);
+    co_return level.value();
+}
+
+ss::future<bool>
+sharded_store::set_compatibility(compatibility_level compatibility) {
+    auto map = [compatibility](store& s) {
+        return s.set_compatibility(compatibility).value();
+    };
+    auto reduce = std::logical_and<>{};
+    co_return co_await _store.map_reduce0(map, true, reduce);
+}
+
+ss::future<bool> sharded_store::set_compatibility(
+  const subject& sub, compatibility_level compatibility) {
+    using overload_t = result<bool> (store::*)(
+      const subject&, compatibility_level);
+    auto set = co_await _store.invoke_on(
+      shard_for(sub),
+      static_cast<overload_t>(&store::set_compatibility),
+      sub,
+      compatibility);
```

Comment:
what's happening here compared to passing a lambda that captures sub/compatibility and invokes `s.set_compatiblity`?

Reply:
> what's happening here compared to passing a lambda that captures sub/compatibility and invokes `s.set_compatiblity`?

It works on g++ and I don't have to name the lambda. The `static_cast` is to disambiguate the `set_compatibility` overloads.

Reply:
got it. let's make sure to mark these with a reference that is searchable so we can go fix these when gcc is fixed. for this particular case you might also put a little note like: just replace this fancy unnamed thing with a lambda.

Reply:
https://github.com/vectorizedio/redpanda/issues/891#issuecomment-879085586